crates/cxx-qt-gen/src/syntax/implitemfn.rs

```diff
@@ -0,0 +1,69 @@
+// SPDX-FileCopyrightText: 2022 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
+// SPDX-FileContributor: Andrew Hayzen <andrew.hayzen@kdab.com>
+//
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+use syn::{FnArg, Receiver, Signature};
+
+use super::types::is_pin_mut;
+
+/// Return whether the first parameter of a method is a mutable self argument
+pub fn is_method_mutable(signature: &Signature) -> bool {
+    match signature.inputs.first() {
+        Some(FnArg::Receiver(Receiver { mutability, ty, .. })) => {
+            // Check if Pin<T> is mut or fallback to normal mutability
+            is_pin_mut(ty) || mutability.is_some()
```

Comment:
We could use this change to fix some of the TODO's in the earlier code by changing this code to also check whether the receiver is a pin of self:
```suggestion
            (is_pin_of_self(ty) && is_pin_mut(ty)) || mutability.is_some()
```

Reply:
Ah an it is using `Receiver` to find if it's a `self`

Reply:
Done